AccountManagementService.Domain/CommandHandlers/CreateTransactionCommandHandler.cs

```diff
@@ -0,0 +1,99 @@
+using AccountManagementService.Domain.Commands;
+using AccountManagementService.Domain.Contracts;
+using AccountManagementService.Domain.Models;
+using AccountManagementService.Domain.Notifications;
+using FluentValidation;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace AccountManagementService.Domain.CommandHandlers;
+
+public class CreateTransactionCommandHandler : IRequestHandler<CreateTransactionCommand, object>
+{
+    private readonly IMediator _mediator;
+    private readonly ITransactionRepository _repository;
+    private readonly IAccountRepository _accountRepository;
+    private readonly IValidator<CreateTransactionCommand> _validator;
+
+    public CreateTransactionCommandHandler(IMediator mediator, ITransactionRepository repository, IAccountRepository accountRepository, IValidator<CreateTransactionCommand> validator)
+    {
+        _mediator = mediator;
+        _repository = repository;
+        _accountRepository = accountRepository;
+        _validator = validator;
+    }
+    
+    public async Task<object> Handle(CreateTransactionCommand request, CancellationToken cancellationToken)
+    {
+
+        try
+        {
+            await _validator.ValidateAndThrowAsync(request, cancellationToken);
+            
+            var transaction = new Transaction(request.TransactionType, request.Amount, request.CreditAccountId, request.DebitAccountId);
+            
+            await _repository.AddTransaction(transaction);
+            await _repository.Save();
+
+            if (request.CreditAccountId != null)
+            {
+                var creditAccount = await _accountRepository.GetAccountById(request.CreditAccountId);
+                if (creditAccount != null)
+                {
+                    if (!creditAccount.IsActive)
```

Comment:
Temos essa forma também, mais pra compartilhar msm <3
```suggestion
                    if (creditAccount.IsActive is false)
```